doc/smart-switch/reboot/reboot-hld.md

```diff
@@ -0,0 +1,337 @@
+
+```
+rpc RebootStatus(RebootStatusRequest) returns (RebootStatusResponse) {}
+
+message RebootStatusRequest {
+  repeated types.Path subcomponents = 1; // optional sub-component.
+}
+
+message RebootStatusResponse {
+  bool active = 1;      // If reboot is active.
+  uint64 wait = 2;      // Time left until reboot.
+  uint64 when = 3;      // Time to reboot in nanoseconds since the epoch.
+  string reason = 4;    // Reason for reboot.
+  uint32 count = 5;     // Number of reboots since active.
+  RebootMethod method = 6; // Type of reboot.
+  RebootStatus status = 7; // Applicable only when active = false.
+}
+
+message RebootStatus {
+  enum Status {
+    STATUS_UNKNOWN = 0;
+    STATUS_SUCCESS = 1;
+    STATUS_RETRIABLE_FAILURE = 2;
+    STATUS_FAILURE = 3;
+  }
+  Status status = 1;
+  string message = 2;
+}
+```
+
+### reboot CLI modifications ###
+
+Introduce a new parameter <span style="color:blue">'-d'</span> to the reboot command for specifying the DPU module name requiring a reboot. If the chassis is
+not a smart switch, this action will have no effect. If the reboot command is executed without specifying any '-d' option, the entire switch will
+be rebooted.
+
+```
+Usage /usr/local/bin/reboot [options]
+    Request rebooting the device. Invoke platform-specific tool when available.
+    This script will shutdown syncd before rebooting.
+
+    Available options:
+        -h, -? : getting this help
+        -f     : execute reboot force
+        -d     : DPU module name
+```
+
+### reboot.py script modifications ###
+
+* Within the reboot() function, incorporate a verification step to invoke is_smartswitch(). Should is_smartswitch() yield false, proceed with the current
+implementation. However, if is_smartswitch() returns true, invoke the new reboot_smartswitch() function, passing a parameter to specify whether it's
+a complete switch reboot or targeting a specific DPU.
+
+* If the reboot_type is ‘REBOOT_TYPE_WARM’ and is_smartswitch is true, return a warning that this type of reboot is not supported.
+
+* Add a new reboot_smartswitch() function to reboot either the entire switch or a particular DPU, which takes DPU ID as an argument that
+needs a reboot.
+
+```
+def reboot_smartswitch(duthost, localhost, reboot_type='cold', reboot_dpu='false', dpu_id='0')
+    """
+    reboots SmartSwitch or a DPU
+    :param duthost: DUT host object
+    :param localhost:  local host object
+    :param reboot_type: reboot type (cold)
+    :param reboot_dpu: reboot dpu or switch (true, false)
+    :param dpu_id: reboot the dpu with id, valid only if reboot_dpu is true.
+```
+
+### Error handling and exception scenarios ###
+
+The following are specific error scenarios where the DPU state will not be DPU_READY.
+
+* If the GNMI service is not operational on the DPU or DPU is unreachable for any reason, detach the PCI, and proceed with the reboot after a timeout
```

Comment:
GNOI service is separate from GNMI server on SmartSwitch, I think we are looking for operational GNOI service